# LetsGrowMore - Data Science intern

## Task 1- Iris Flower Classification ML Project

###  Author - Simran Sayyed


   ### Loading the data set

In [25]:
import numpy as np
import seaborn as sns
print(sns.get_dataset_names())
df=sns.load_dataset('iris')
df

['anagrams', 'anscombe', 'attention', 'brain_networks', 'car_crashes', 'diamonds', 'dots', 'dowjones', 'exercise', 'flights', 'fmri', 'geyser', 'glue', 'healthexp', 'iris', 'mpg', 'penguins', 'planets', 'seaice', 'taxis', 'tips', 'titanic']


sepal_length  sepal_width  petal_length  petal_width    species
0             5.1          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[150 rows x 5 columns]

### Checking for missing values

In [26]:
df.isnull().sum()

sepal_length    0
sepal_width     0
petal_length    0
petal_width     0
species         0
dtype: int64

### Label encoding

In [27]:
##label encoding
from sklearn import preprocessing
le=preprocessing.LabelEncoder()
le.fit(df["species"])

LabelEncoder()

In [28]:

df["species"]=le.transform(df["species"])
df

sepal_length  sepal_width  petal_length  petal_width  species
0             5.1          3.5           1.4          0.2        0
1             4.9          3.0           1.4          0.2        0
2             4.7          3.2           1.3          0.2        0
3             4.6          3.1           1.5          0.2        0
4             5.0          3.6           1.4          0.2        0
..            ...          ...           ...          ...      ...
145           6.7          3.0           5.2          2.3        2
146           6.3          2.5           5.0          1.9        2
147           6.5          3.0           5.2          2.0        2
148           6.2          3.4           5.4          2.3        2
149           5.9          3.0           5.1          1.8        2

[150 rows x 5 columns]

### Data Separation

In [29]:

x=df.iloc[:,0:4]
x

sepal_length  sepal_width  petal_length  petal_width
0             5.1          3.5           1.4          0.2
1             4.9          3.0           1.4          0.2
2             4.7          3.2           1.3          0.2
3             4.6          3.1           1.5          0.2
4             5.0          3.6           1.4          0.2
..            ...          ...           ...          ...
145           6.7          3.0           5.2          2.3
146           6.3          2.5           5.0          1.9
147           6.5          3.0           5.2          2.0
148           6.2          3.4           5.4          2.3
149           5.9          3.0           5.1          1.8

[150 rows x 4 columns]

In [30]:

y=df.iloc[:,-1]
y

0      0
1      0
2      0
3      0
4      0
      ..
145    2
146    2
147    2
148    2
149    2
Name: species, Length: 150, dtype: int32

### Train Test Split

In [31]:

from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2)
x_train

sepal_length  sepal_width  petal_length  petal_width
111           6.4          2.7           5.3          1.9
57            4.9          2.4           3.3          1.0
30            4.8          3.1           1.6          0.2
65            6.7          3.1           4.4          1.4
106           4.9          2.5           4.5          1.7
..            ...          ...           ...          ...
14            5.8          4.0           1.2          0.2
56            6.3          3.3           4.7          1.6
77            6.7          3.0           5.0          1.7
36            5.5          3.5           1.3          0.2
34            4.9          3.1           1.5          0.2

[120 rows x 4 columns]

## KNN Classifier

In [32]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score
knn=KNeighborsClassifier(n_neighbors=10)
knn.fit(x_train,y_train)
y_pred=knn.predict(x_test)


### Accuracy

In [33]:
acc1=accuracy_score(y_test,y_pred)
acc1

0.9666666666666667

### Confusion Matrix

In [34]:
from sklearn.metrics import confusion_matrix
c_m=confusion_matrix(y_test,y_pred)
c_m 

array([[11,  0,  0],
       [ 0, 10,  1],
       [ 0,  0,  8]], dtype=int64)

### Classification Report

In [35]:
from sklearn.metrics import classification_report
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        11
           1       1.00      0.91      0.95        11
           2       0.89      1.00      0.94         8

    accuracy                           0.97        30
   macro avg       0.96      0.97      0.96        30
weighted avg       0.97      0.97      0.97        30



## Random Forest Classifier

In [36]:
from sklearn.ensemble import RandomForestClassifier
clf=RandomForestClassifier(n_estimators=10)
clf.fit(x_train,y_train)
y_pred=clf.predict(x_test)


### Accuracy

In [37]:
acc2=accuracy_score(y_test,y_pred)
acc2

0.9

### Confusion Matrix

In [38]:
cm=confusion_matrix(y_test,y_pred)
cm

array([[11,  0,  0],
       [ 0, 11,  0],
       [ 0,  3,  5]], dtype=int64)

### Classification Report

In [39]:
from sklearn.metrics import classification_report
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        11
           1       0.79      1.00      0.88        11
           2       1.00      0.62      0.77         8

    accuracy                           0.90        30
   macro avg       0.93      0.88      0.88        30
weighted avg       0.92      0.90      0.89        30



## Conclusion:
Iris Flower Classification has been done using two classifiers:

Random Forest Classifier and
KNN Classifier.

Accuracy of KNN Classifier is 96%

Accuracy of Random Forest Classifier is 90%

Thus in conclusion,KNN Classifier is better for this classification .